scripts/docker-run.sh

```diff
@@ -43,7 +43,7 @@ docker build \
 # Note that the `--privileged` flags is set by default because it is
 # required to register `binfmt` handlers, whaich allow to run builders
 # for ARM achitectures which need to use `qemu-user`.
-docker_flags=("--privileged" "--interactive")
+docker_flags=("--rm" "--privileged" "--interactive")
```

Comment:
Might be good to add a comment on why we need this above, as well.

Reply:
Could the `--privileged` possibly be scoped down to only those containers that need it? For example, it's completely unnecessary (or at least I *think* it should should be) for running the `make generate` step through docker.

Reply:
> Could the --privileged possibly be scoped down to only those containers that need it? For example, it's completely unnecessary (or at least I think it should should be) for running the make generate step through docker.

I mode `--priveleged` disabled by default. When it is needed, it can be activated by setting `DOCKERP_PRIVILEGED` environment variable to `true`.